// dbProxy/dbProxy.js
var config = require('config');
var logger = require("logger");
var mongo = require("mongodb");

var settings = {
    w: 1,
    journal: false,
    fsync: false
};


var db = new mongo.Db(config.db.source, new mongo.Server(config.db.host, config.db.port, {
    safe: true
}), settings);

var init = function () {
    db.open(function (err) {
        if (err) return logger.error(err);
        //logger.info('成功连接数据库');
    });
};

exports.find = function (name, filter, fn) {
    db.collection(name, function (err, collection) {
        if (err) return fn(err);
        collection.find(filter, fn);
    });
};

exports.findOne = function (name, filter, fn) {
    if (typeof (filter) == "function") {
        fn = filter;
        filter = null;
    }
    db.collection(name, function (err, collection) {
        if (err) return fn(err);
        collection.findOne(filter, fn);
    });
};

exports.insert = function (name, data, fn) {
    db.collection(name, function (err, collection) {
        if (err) return fn(err);
        collection.insert(data, fn);
    });
};

exports.update = function (name, filter, set, fn) {
    db.collection(name, function (err, collection) {
        if (err) return fn(err);
        collection.update(filter, { $set: set }, fn);
    });
};

init();

exports.db = db;
