// cookie/cookie.js
var querystring = require('querystring');

var req;
var res;
var encode = encodeURIComponent;
var decode = decodeURIComponent;

exports.body = {};

exports.init = function (request, response) {
    req = request;
    res = response;
    if (req.headers && req.headers.cookie)
        exports.body = querystring.parse(req.headers.cookie, '; ');
}

exports.remove = function (name) {
    var opt = {
        expires: 0,
        maxAge: 0,
        httpOnly: true
    }
    var cookie = serialize(name, '', opt);
    res.setHeader('Set-Cookie', cookie);
};

exports.set = function (name, value, opt) {
    if (opt === null) {
        opt = 3600;
    }
    if (opt === 'number') {
        opt = {
            httpOnly: true,
            maxAge: opt
        }
    }

    if (!name || !value)
        throw new Error('名称和值不允许为空');

    var cookie = serialize(name, value, opt);
    res.setHeader('Set-Cookie', cookie);
}

var serialize = function (name, val, opt) {
    var pairs = [name + '=' + encode(val)];
    opt = opt || {};

    if (opt.maxAge)
        pairs.push('Max-Age=' + opt.maxAge);
    if (opt.domain)
        pairs.push('Domain=' + opt.domain);
    if (opt.path)
        pairs.push('Path=' + opt.path);
    if (opt.expires)
        pairs.push('Expires=' + opt.expires.toUTCString());
    if (opt.httpOnly)
        pairs.push('HttpOnly');
    if (opt.secure)
        pairs.push('Secure');

    return pairs.join('; ');
};
