// unload/unload.js
var config = require('config');
var fs = require('fs');
var logger = require('logger');
var path = require('path');

var exec = function (bus) {
    bus.res.writeHead(200, { 'content-type': 'application/json;charset=utf-8' });
    bus.res.end(JSON.stringify(bus.data));
    logger.info('成功返回[' + ((new Date - bus.startTime)) + 'ms]');
};

exports.exec = exec;

exports.execError = function (bus, msg) {
    msg = msg ? msg : "操作失败";
    bus.data.code = config.code.error;
    bus.data.msg = msg + '';
    exec(bus);
};
exports.execSuccess = function (bus, msg) {
    msg = msg ? msg : "操作成功";
    bus.data.code = config.code.success,
    bus.data.msg = msg + '';
    exec(bus);
};
exports.execFile = function (bus) {
    fs.readFile(bus.data.file, function (error, data) {
        if (error) {
            bus.data.error = error;
            return execError(bus);
        }
        var ext = path.extname(bus.data.file);
        ext = ext ? ext.slice(1) : 'unknown';
        var contentType = config.mime[ext] || "text/plain";
        bus.res.writeHead(200, { 'Content-Type': contentType + ';charset=utf-8' });
        bus.res.write(data, "binary");
        bus.res.end();
        var now = new Date;
        logger.info('成功返回(' + bus.data.file + ')[' + ((now - bus.startTime)) + 'ms]');
    });
}

exports.execFileSync = function (bus) {
    logger.debug(bus.data.file);
    var data = fs.readFileSync(bus.data.file);

    var ext = path.extname(bus.data.file);
    ext = ext ? ext.slice(1) : 'unknown';
    var contentType = config.mime[ext] || "text/plain";
    var now = new Date;
    logger.info('成功返回(' + bus.data.file + ')[' + ((now - bus.startTime)) + 'ms]');
    bus.res.writeHead(200, { 'Content-Type': contentType + ';charset=utf-8' });
    bus.res.write(data);
    bus.res.end();
}

//exports.init = function (bus) {
//    res = bus.res;
//    startTime = bus.startTime;
//};

//var factory = function () {
//    //
//};
